### PR TITLE
Changes from review remote worker

### PR DESCRIPTION
**Pull request by wesleymonte**

### CODE REVIEW

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java [hunk 1]
```diff
@@ -11,44 +12,49 @@
     private static final Logger LOGGER = Logger.getLogger(DockerCommandExecutor.class);
     private final WorkerDockerRequestHelper workerDockerRequestHelper;
 
+    private final long poolingPeriodTime = 300;
```
Comment (joivo) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java]: Replace the word 'time' for the corresponding unit of measure.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java [hunk 2]
```diff
@@ -11,44 +12,49 @@
...
     }
 
-    public void asyncExecuteCommand(String command, String taskId) throws Exception {
-        LOGGER.info("Sending command to the [" + command + "] for the task [" + taskId + "]"
-                + this.workerDockerRequestHelper.getContainerName() + "].");
+    /**
+     * It creates the command execution instance and sends it to the docker, without waiting for the
+     * end of execution and nor for its exit code.
+     */
+    public void asyncExecuteCommand(String command) throws Exception {
```
Comment (joivo) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java]: Replace by `executeAsyncCommand`.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerExecutorHelper.java [hunk 3]
```diff
@@ -17,80 +17,83 @@
...
+    public DockerExecutorHelper(String taskScriptContent,
+        WorkerDockerRequestHelper workerDockerRequestHelper) {
         this.taskScriptContent = taskScriptContent;
         this.workerDockerRequestHelper = workerDockerRequestHelper;
         this.dockerCommandExecutor = new DockerCommandExecutor(workerDockerRequestHelper);
     }
 
-    public void runScriptExecutor(String taskId, String tsFilepath) throws Exception {
-        this.dockerCommandExecutor.asyncExecuteCommand("/bin/bash " + TASK_SCRIPT_EXECUTOR + " -d -tsf=" + tsFilepath, taskId);
+    public void runScriptExecutor(String tsFilepath) throws Exception {
```
Comment (joivo) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerExecutorHelper.java]: Replace by `runExecutorScript`.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerExecutorHelper.java [hunk 4]
```diff
@@ -17,80 +17,83 @@
...
+        LOGGER.info("Running script executor to task file [" + tsFilepath + "].");
+        this.dockerCommandExecutor
+            .asyncExecuteCommand("/bin/bash " + TASK_SCRIPT_EXECUTOR + " -d -tsf=" + tsFilepath);
     }
 
-    public void sendTaskScriptExecutor(String taskId) throws Exception {
+    public void sendTaskScriptExecutor() throws Exception {
```
Comment (joivo) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerExecutorHelper.java]: Replace by `sendTaskExecutorScript`.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java [hunk 5]
```diff
@@ -4,157 +4,223 @@
...
+ * execution a new container is started and after the execution of the task is finished the
+ * container is removed. The execution of each command is not done directly by this class but by a
+ * bash script sent into the container. In cases where container initialization error occurs, the
+ * commands go to failure state and a failure result is returned.
+ */
 public class DockerTaskExecutor implements TaskExecutor {
 
-    private static final int SUCCESS_EXIT_CODE = 0;
-    private static final int FAIL_EXIT_CODE = 127;
+    private static final long poolingPeriodTime = 2000;
```
Comment (joivo) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java]: Replace the word 'time' for the corresponding unit of measure.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java [hunk 6]
```diff
@@ -4,157 +4,223 @@
...
-        List<Command> commands = task.getTaskSpec().getCommands();
-        this.dockerExecutorHelper.sendTaskScriptExecutor(task.getId());
-        String taskScriptFilepath = "/tmp/" + task.getId() + ".ts";
-        this.dockerExecutorHelper.sendTaskScript(commands, taskScriptFilepath,
-            task.getId());
-        this.dockerExecutorHelper.runScriptExecutor(task.getId(), taskScriptFilepath);
+    /**
+     * Sends the executor task script and write task commands inside the .ts file.
+     */
+    private String containerEnvironmentSetup(String taskId, List<Command> commands)
```
Comment (joivo) [inline src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java]: Replace by 'setupContainerEnviroment'.

**Review by joivo — CHANGES_REQUESTED** [review]

**Comment by joivo** [comment]
Commit these changes in the right branch (`feature / remote-worker`) not messing the repository story.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java (hunk 1 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java]: Done.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java (hunk 2 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerCommandExecutor.java]: Done.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerExecutorHelper.java (hunk 3 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerExecutorHelper.java]: Done.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerExecutorHelper.java (hunk 4 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerExecutorHelper.java]: Done.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java (hunk 5 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java]: Done.

File: src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java (hunk 6 above)
Comment (wesleymonte) [reply src/main/java/org/fogbowcloud/arrebol/execution/docker/DockerTaskExecutor.java]: Done.

**Review by joivo — APPROVED** [review]

**Pull request merged**